django/db/models/sql/compiler.py

```diff
@@ -150,7 +147,14 @@ def collapse_group_by(self, expressions, having):
                         getattr(expr.output_field, 'model') == self.query.model):
                     pk = expr
             if pk:
+                # MySQLism: Columns in HAVING clause must be added to the GROUP BY.
                 expressions = [pk] + [expr for expr in expressions if expr in having]
+        elif self.connection.features.allows_group_by_selected_pks:
+            pks = {expr for expr in expressions if hasattr(expr, 'target') and expr.target.primary_key}
+            aliases = {expr.alias for expr in pks}
+            expressions = [
+                expr for expr in expressions if expr in pks or getattr(expr, 'alias', None) not in aliases
```

Comment:
From a quick investigation (running `aggregation_regress` with `expr.alias not in aliases`) the following expression with no alias end up here:

1. `CombinedExpression`
2. `RawSQL`
3. `Date`

> It might be possible to fix this in a way that only alias.col references end up in here in the first place.

Should we tackle it in this patch?